crates/compiler/fmt/src/expr.rs

```diff
@@ -618,6 +618,22 @@ fn starts_with_newline(expr: &Expr) -> bool {
     }
 }
 
+fn fmt_str_body(body: &str, buf: &mut Buf) {
+    for c in body.chars() {
+        match c {
+            // Format blank characters as unicode escapes
+            '\u{200a}' => buf.push_str("\\u(200a)"),
+            '\u{200b}' => buf.push_str("\\u(200b)"),
+            '\u{200c}' => buf.push_str("\\u(200c)"),
+            '\u{feff}' => buf.push_str("\\u(feff)"),
+            // Don't change anything else in the string
+            ' ' => buf.spaces(1),
+            '\n' => buf.newline(),
```

Comment:
I think for these, out of an abundance of caution, we should use `buf.push_str_allow_spaces(" ")` and `buf.push_str_allow_spaces("\n")`. (or equivalents - feel free to refactor!)

The only case in which that would matter is if the string is multi-line and contains trailing whitespace. The spaces()/newline() functions will automatically strip that, which is probably not what we want for the contents of a string.

Reply:
Great point!  I made those changes and also added a test for this case here: 1551d8fb0da07f2fb03e74ae2c2300b44dd02b75